homeassistant/components/http/auth.py

```diff
@@ -223,6 +226,20 @@ async def auth_middleware(
             authenticated = True
             auth_type = "signed request"
 
+        ip_address_ = ip_address(request.remote)  # type: ignore[arg-type]
+        if (
+            not authenticated
+            and strict_connection
+            and (is_cloud_connection(hass) or not is_local(ip_address_))
+        ):
+            if not (transport := request.transport):
+                # If we don't have a transport, we can't close the connection.
+                # This should never happen.
+                raise HTTPBadRequest()
```

Comment:
If this can't happen, we should just assert transport is not None or cast it to the expected type

Reply:
I was worried about any potential for DOS or similar and didn't want to leave a possibility of crashing here. I'm not very familiar with web servers.